src/Stackctl/Config/RequiredVersion.hs

```diff
@@ -18,19 +19,24 @@ import qualified Data.Version as Version
 import Text.ParserCombinators.ReadP (readP_to_S)
 
 data RequiredVersion = RequiredVersion
-  { requiredVersionOp :: Text
-  , requiredVersionCompare :: Version -> Version -> Bool
+  { requiredVersionOp :: RequiredVersionOp
   , requiredVersionCompareWith :: Version
   }
-
-instance Show RequiredVersion where
-  show RequiredVersion {..} =
-    unpack requiredVersionOp <> " " <> showVersion requiredVersionCompareWith
+  deriving stock (Eq, Show)
 
 instance FromJSON RequiredVersion where
   parseJSON =
     withText "RequiredVersion" $ either fail pure . requiredVersionFromText
 
+instance ToJSON RequiredVersion where
```

Comment:
Because we parse `"1.2.3.4"` as `"= 1.2.3.4"`, such a value wouldn't round-trip in `toJSON . parseJSON`. Not a big deal, IMO, but that'd probably bite someone if they wanted to add the usual QuickCheck property on that.

Reply:
I'm not sure I understand the suggestion. If you think it's worth exploring, can you expand or rephrase?

Reply:
I decided to add the round-trip test. These need to be `decode $ encode {value}`, which doesn't suffer from the gotcha. Only an `encode $ decode <text>` test would, which you can't write anyway since you don't want to be using arbitrary _text_ for these tests.